src/deepsparse/utils/timer.py

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from dataclasses import dataclass
+from typing import Dict, List
+
+
+__all__ = ["InferenceStages", "StagedTimer", "InferenceTimer"]
+
+
+@dataclass(frozen=True)
+class InferenceStages:
+    PRE_PROCESS: str = "pre_process"
+    ENGINE_FORWARD: str = "engine_forward"
+    POST_PROCESS: str = "post_process"
+    TOTAL_INFERENCE: str = "total_inference"
+
+
+class StagedTimer:
```

Comment:
Possibly some inheritance between `Timers` would lead to better code efficiency?

Reply:
now split out into a single timer and a manger